src/airdrop/src/lib.rs

```diff
@@ -334,5 +336,67 @@ fn bring_caninster_back_to_life() -> CustomResult<()> {
     Ok(())
 }
 
+/// Returns all the eth addresses with how much is meant to be sent to each one of them
+#[update(guard = "caller_is_admin")]
+pub fn get_airdrop(mut index: Index) -> CustomResult<(Index, Vec<EthAddressAmount>)> {
+    check_if_killed()?;
+
+    read_state(|state| {
+        let mut airdrop = Vec::default();
+
+        // TODO: dangerous casting from u64 to usize?
+        for idx in (index.0 as usize)..state.airdrop_reward.len() {
+            if !state.airdrop_reward[idx].transferred {
+                airdrop.push(state.airdrop_reward[idx].clone());
+                index = Index(idx as u64);
+            }
+        }
+
+        Ok((index, airdrop))
+    })
+}
+
+/// Pushes the new state of who was transferred money
+#[update(guard = "caller_is_admin")]
+pub fn put_airdrop(index: Index, eth_address_amount: EthAddressAmount) -> CustomResult<()> {
+    check_if_killed()?;
+
+    mutate_state(|state| {
+        // for the index to the end of the list update the state
+        for i in index.0..state.airdrop_reward.len() as u64 {
+            if state.airdrop_reward[i as usize].eth_address == eth_address_amount.eth_address {
+                state.airdrop_reward[i as usize].transferred = true;
+            }
+        }
+    });
+
+    Ok(())
+}
+
+/// For a given principal check if the original airdrop amount has been redeemed
+#[query]
+pub fn has_redeemed() -> CustomResult<bool> {
+    check_if_killed()?;
+
+    let caller_principal = caller();
+
+    read_state(|state| {
+        // get the code and eth_address associated with the principal
+        let (_, eth_address) = state
```

Comment:
this duplicates the code on line 295, worth a utility?

Reply:
No, not if we integrate it with get_code as we talked about previously!

Reply:
Cool then